src/js/tracker.js

```diff
@@ -1604,25 +1563,70 @@
 				helpers.addEventListener(windowAlias, 'focus', activityHandler);
 				helpers.addEventListener(windowAlias, 'blur', activityHandler);
+			}
 
+			if (activityTrackingConfig.enabled && (resetActivityTrackingOnPageView || installingActivityTracking)) {
 				// Periodic check for activity.
 				lastActivityTime = now.getTime();
-				clearInterval(pagePingInterval);
-				pagePingInterval = setInterval(function heartBeat() {
-					var now = new Date();
-
-					// There was activity during the heart beat period;
-					// on average, this is going to overstate the visitDuration by configHeartBeatTimer/2
-					if ((lastActivityTime + configHeartBeatTimer) > now.getTime()) {
-						// Send ping if minimum visit time has elapsed
-						if (configMinimumVisitTime < now.getTime()) {
-							logPagePing(finalizeContexts(context, contextCallback)); // Grab the min/max globals
-						}
+
+				for (var key in activityTrackingConfig.configurations) {
+					if (activityTrackingConfig.configurations.hasOwnProperty(key)) {
+						const config = activityTrackingConfig.configurations[key];
+
+						//Clear page ping heartbeat on new page view
+						clearInterval(config.activityInterval);
+
+						config.activityInterval = activityInterval({
+							...config,
+							configLastActivityTime: lastActivityTime,
+							context: finalizeContexts(context, contextCallback)
+						});
 					}
-				}, configHeartBeatTimer);
+				}
 			}
 		}
 
+		function activityInterval({ configHeartBeatTimer, configMinimumVisitLength, configLastActivityTime, callback, context }) {
+			return setInterval(function heartBeat() {
+				var now = new Date();
+
+				// There was activity during the heart beat period;
+				// on average, this is going to overstate the visitDuration by configHeartBeatTimer/2
+				if ((lastActivityTime + configHeartBeatTimer) > now.getTime()) {
+					// Send ping if minimum visit time has elapsed
+					if (configLastActivityTime + configMinimumVisitLength * 1000 < now.getTime()) {
+						refreshUrl();
+						callback({ context, pageViewId: getPageViewId(), minXOffset, minYOffset, maxXOffset, maxYOffset });
+						resetMaxScrolls();
+					}
+				}
+			}, configHeartBeatTimer);
+		}
+
+		/**
+		 * Configured the activity tracking and
+		 * ensures good values for min visit and heartbeat
+		 *
+		 * @param minimumVisitLength string The minimum length of a visit before the first page ping
+		 * @param heartBeatDelay string The length between checks to see if we should send a page ping
+		 * @param callback function A callback function to execute
+		 */
+		function configureActivityTracking(minimumVisitLength, heartBeatDelay, callback) {
+			if (minimumVisitLength === parseInt(minimumVisitLength, 10) &&
+				heartBeatDelay === parseInt(heartBeatDelay, 10)) {
+				return {
+					activityTrackingEnabled: true,
+					configMinimumVisitLength: minimumVisitLength,
+					configHeartBeatTimer: heartBeatDelay * 1000,
```

Comment:
I haven't tested this, but it looks like you are using the string version of heartBeatDelay instead of the numeric version you got by `parseInt`

Also, I think `activityTrackingEnabled` never gets used.  You only ever use the `enabled` property on the parent object.

Reply:
My comment in nonsense on this function. heartBeatDelay should always be an `int` (see apiMethods that call this for clarification). I've updated the comment.